src/context/directory/handlers/hooks.js

```diff
@@ -0,0 +1,58 @@
+import fs from 'fs-extra';
+import path from 'path';
+import { constants } from 'auth0-source-control-extension-tools';
+
+import { getFiles, existsMustBeDir, loadJSON, sanitize } from '../../../utils';
+import log from '../../../logger';
+
+
+function parse(context) {
+  const hooksFolder = path.join(context.filePath, constants.HOOKS_DIRECTORY);
+  if (!existsMustBeDir(hooksFolder)) return { hooks: [] }; // Skip
+
+  const files = getFiles(hooksFolder, [ '.json' ]);
+
+  const hooks = files.map((f) => {
+    const hook = { ...loadJSON(f, context.mappings) };
+    if (hook.script) {
+      hook.script = context.loadFile(hook.script, constants.HOOKS_DIRECTORY);
+    }
+
+    hook.name = hook.name.toLowerCase().replace(/\s/g, '-');
+
+    return hook;
+  });
+
+  return {
+    hooks
+  };
+}
+
+async function dump(context) {
+  const hooks = [ ...context.assets.hooks || [] ];
+
+  if (!hooks) return; // Skip, nothing to dump
```

Comment:
Probably not a huge deal since you are iterating below and it will end up almost a noop, but this will always evaluate to false since you are initializing in an array above. Fixing this will prevent the unneeded fs call below though.

You need to check on the length here instead.